### PR TITLE
Do not unwrap with new type. 

### PR DESCRIPTION
**Pull request by larry-aptos**
* devnet build at  4318127b6848cbdf901834706258a42f7c3da414

* Adds metrics to record if there is incompatibility.  

## Test plan

* Deployed to devnet and looks good.

* Also tested locally, looks good(tested from version `472207`)

image

### CODE REVIEW

**Comment by bowenyang007** [comment]
Caveat is that this PR will skip some transactions.

**Review by bowenyang007 — APPROVED** [review]

File: rust/processor/src/processors/token_v2_processor.rs [hunk 1]
```diff
@@ -454,7 +454,7 @@ async fn parse_v2_token(
 
     // Code above is inefficient (multiple passthroughs) so I'm approaching TokenV2 with a cleaner code structure
     for txn in transactions {
-        let txn_data = txn.txn_data.as_ref().expect("Txn Data doesn't exit!");
+        let txn_data = txn.txn_data.as_ref().expect("Txn Data doesn't exist!");
```
Comment (rtso) [inline rust/processor/src/processors/token_v2_processor.rs:457]: can we also log the txn version number?

File: rust/processor/src/processors/user_transaction_processor.rs [hunk 2]
```diff
@@ -136,7 +136,7 @@ impl ProcessorTrait for UserTransactionProcessor {
         for txn in transactions {
             let txn_version = txn.version as i64;
             let block_height = txn.block_height as i64;
-            let txn_data = txn.txn_data.as_ref().expect("Txn Data doesn't exit!");
+            let txn_data = txn.txn_data.as_ref().expect("Txn Data doesn't exist!");
```
Comment (rtso) [inline rust/processor/src/processors/user_transaction_processor.rs:139]: same here

Comment (larry-aptos) [reply rust/processor/src/processors/user_transaction_processor.rs]: removed; change to match.

File: rust/processor/src/processors/token_v2_processor.rs (hunk 1 above)
Comment (larry-aptos) [reply rust/processor/src/processors/token_v2_processor.rs]: oh i see, working on it.

Comment (larry-aptos) [reply rust/processor/src/processors/token_v2_processor.rs]: done.

File: rust/processor/src/processors/token_v2_processor.rs [hunk 3]
```diff
@@ -453,13 +453,10 @@ async fn parse_v2_token(
...
-        let txn_data = txn.txn_data.as_ref().unwrap_or_else(|| {
-            error!(
-                transaction_version = txn.version,
-                "Txn Data doesn't exist for version {}", txn.version
-            );
-            panic!();
-        });
+        let txn_data = match txn.txn_data.as_ref() {
+            Some(data) => data,
+            None => continue,
```
Comment (rtso) [inline rust/processor/src/processors/token_v2_processor.rs:458]: Missing the metric and log here?

Comment (larry-aptos) [reply rust/processor/src/processors/token_v2_processor.rs]: Done.

File: rust/processor/src/models/account_transaction_models/account_transactions.rs [hunk 4]
```diff
@@ -42,13 +41,19 @@ impl AccountTransaction {
...
-            );
-            panic!();
-        });
+        let txn_data = match transaction.txn_data.as_ref() {
+            Some(data) => data,
+            None => {
+                PROCESSOR_UNKNOWN_TYPE_COUNT
+                    .with_label_values(&["AccountTransaction"])
+                    .inc();
+                tracing::warn!(
```
Comment (just-in-chang) [inline rust/processor/src/models/account_transaction_models/account_transactions.rs:50]: Maybe helpful to keep `transaction_version = transaction.version` for structured log? (same as others below)

**Review by just-in-chang — APPROVED** [review]

File: rust/processor/src/models/account_transaction_models/account_transactions.rs (hunk 4 above)
Comment (larry-aptos) [reply rust/processor/src/models/account_transaction_models/account_transactions.rs:50]: good call; fixed now.

**Pull request merged**